File: 프로그래머스/lv1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] arr) {
        if (arr.length == 1) {
            int[] answer = {-1};
            return answer;
        }
        
        int[] answer = new int[arr.length - 1];
		int j = 0;
        int min = arr[0];
        for(int i = 0; i < arr.length; i++) {
            min = Math.min(min, arr[i]);
        }
        
		 for(int i = 0; i < arr.length; i++) {
            if(arr[i] == min) {
                continue;
            }
            
            answer[j++] = arr[i];
        }
        return answer;
    }
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.js
function solution(arr) {
    if (arr.length === 1) {
        return [-1];
    }

    let answer = [];
    let min = arr[0];

    // 배열의 최소값을 찾는다
    for (let i = 0; i < arr.length; i++) {
        min = Math.min(min, arr[i]);
    }

    // 최소값을 제외한 배열을 생성한다
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === min) {
            continue;
        }
        answer.push(arr[i]);
    }

    return answer;
}
